lib/generator/template_values.rb

```diff
@@ -26,26 +26,30 @@ def template_values
         abbreviated_commit_hash: canonical_data.abbreviated_commit_hash,
         canonical_data_version: canonical_data.version,
         version: version,
-        exercise_name: exercise_name,
+        exercise_name: slug_underscore,
         test_cases: extract
       )
     end
 
     private
 
+    def slug_underscore
+      slug ? slug.tr('-_', '_') : ''
```

Comment:
There is an unnecessary underscore replaced by underscore here.
Also, not sure about the `else ''` we probably should be blowing up if there is no exercise name.